common/src/main/java/com/wynntils/screens/lists/entries/OverlayEntry.java

```diff
@@ -0,0 +1,133 @@
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.screens.OverlayManagementScreen;
+import com.wynntils.screens.lists.OverlayList;
+import com.wynntils.utils.objects.CommonColors;
+import com.wynntils.utils.objects.CustomColor;
+import java.util.List;
+import net.minecraft.client.gui.components.ContainerObjectSelectionList;
+import net.minecraft.client.gui.components.events.GuiEventListener;
+import net.minecraft.client.gui.narration.NarratableEntry;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
+
+public class OverlayEntry extends ContainerObjectSelectionList.Entry<OverlayEntry> {
+    private static final CustomColor ENABLED_COLOR = new CustomColor(0, 116, 0, 255);
+    private static final CustomColor DISABLED_COLOR = new CustomColor(116, 0, 0, 255);
+    private static final CustomColor ENABLED_COLOR_BORDER = new CustomColor(0, 220, 0, 255);
+    private static final CustomColor DISABLED_COLOR_BORDER = new CustomColor(220, 0, 0, 255);
+
+    private static final List<Component> HELP_TOOLTIP_LINES = List.of(
+            new TextComponent("Left click on the overlay to edit it."),
+            new TextComponent("Right click on the overlay to disable/enable it."));
+
+    private final Overlay overlay;
+
+    public OverlayEntry(Overlay overlay) {
+        this.overlay = overlay;
+    }
+
+    @Override
+    public List<? extends NarratableEntry> narratables() {
+        return ImmutableList.of();
+    }
+
+    @Override
+    public void render(
+            PoseStack poseStack,
+            int index,
+            int top,
+            int left,
+            int width,
+            int height,
+            int mouseX,
+            int mouseY,
+            boolean isMouseOver,
+            float partialTick) {
+        poseStack.pushPose();
+        poseStack.translate(left, top, 0);
+
+        boolean enabled = OverlayManager.isEnabled(this.overlay);
+        int y = index != 0 ? 2 : 0;
+
+        RenderUtils.drawRect(
+                poseStack,
+                (enabled ? ENABLED_COLOR_BORDER : DISABLED_COLOR_BORDER).withAlpha(100),
+                0,
+                y,
+                0,
+                width,
+                height - y);
+        RenderUtils.drawRectBorders(poseStack, enabled ? ENABLED_COLOR : DISABLED_COLOR, 0, y, width, height, 1, 2);
+
+        poseStack.translate(0, 0, 1);
+        String translatedName = this.overlay.getTranslatedName();
+        FontRenderer.getInstance()
+                .renderText(
+                        poseStack,
+                        translatedName,
+                        3,
+                        (OverlayList.getItemHeight()
+                                        - FontRenderer.getInstance()
+                                                        .calculateRenderHeight(List.of(translatedName), width)
+                                                / 2f)
+                                / 2f,
```

Comment:
spotless epic fail

Reply:
I agree that spotless handles this anti-gracefullly. But, I think you can also see it as a sign that the function call is too complex. Try extracing the height calculation into a temp variable first. By giving it a name, it will also be clearer what this is supposed to do. A win-win! :)